test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -808,6 +820,416 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, newspec)),
+					),
+				)).ToNot(HaveOccurred())
+
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+				stalldEnabled, rtKernel = true, true
+				noHzParam = fmt.Sprintf("nohz_full=%s", *profile.Spec.CPU.Isolated)
+				sysctlMap = map[string]string{
+					"kernel.hung_task_timeout_secs": "600",
+					"kernel.nmi_watchdog":           "0",
+					"kernel.sched_rt_runtime_us":    "-1",
+					"vm.stat_interval":              "10",
+				}
+
+				kernelParameters = []string{noHzParam, "tsc=nowatchdog", "nosoftlockup", "nmi_watchdog=0", "mce=off", "skew_tick=1",
+					"processor.max_cstate=1", "intel_idle.max_cstate=0", "intel_pstate=disable", "idle=poll"}
+				checkTunedParameters(workerRTNodes, stalldEnabled, sysctlMap, kernelParameters, rtKernel)
+			})
+
+			It("[test_id:54184]Verify enabling both HighPowerConsumption and PerPodPowerManagment fails", func() {
+
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					PerPodPowerManagement: pointer.BoolPtr(true),
+					HighPowerConsumption:  pointer.BoolPtr(true),
+					RealTime:              pointer.BoolPtr(true),
+				}
+				EventuallyWithOffset(1, func() string {
+					err := testclient.Client.Update(context.TODO(), profile)
+					if err != nil {
+						statusErr, _ := err.(*errors.StatusError)
+						return statusErr.Status().Message
+					}
+					return fmt.Sprint("Profile applied successfully")
+				}, time.Minute, 5*time.Second).Should(ContainSubstring("HighPowerConsumption and PerPodPowerManagement can not be both enabled"))
+			})
+
+			It("[test_id:54185] Verify sysfs parameters of guaranteed pod with powersave annotations", func() {
+
+				// This test requires real hardware with powermanagement settings done on BIOS
+				// Using numa nodes to check if we are running on real hardware.
+				for _, node := range workerRTNodes {
+					numaInfo, err := nodes.GetNumaNodes(&node)
+					Expect(err).ToNot(HaveOccurred())
+					if len(numaInfo) < 2 {
+						Skip(fmt.Sprintf("This test need 2 NUMA nodes.The number of NUMA nodes on node %s < 2", node.Name))
+					}
+					// Additional check so that test gets skipped on vm with fake numa
+					onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+					Expect(err).ToNot(HaveOccurred())
+					onlineCPUInt, err := strconv.Atoi(onlineCPUCount)
+					Expect(err).ToNot(HaveOccurred())
+					if onlineCPUInt < 32 {
+						Skip(fmt.Sprintf("This test needs more than 32 CPUs online to work correctly, current online CPUs are %s < 32", onlineCPUCount))
+					}
+				}
```

Comment:
Can you move this to a utility function so it doesn't need to be repeated below? I can imagine other test cases may want to use this in the future as well.

Reply:
Addressed this in latest commit